quicfire_tools/inputs.py

```diff
@@ -2205,81 +2376,449 @@ class QU_metparams(InputFile):
+    primary source for wind speed(s) and direction(s).
+    Multiple sensor*.inp files may be created.
 
     Attributes
     ----------
+    sensor_number : PositiveInt
+        Number representing the wind sensor
     time_now : PositiveInt
         Begining of time step in Unix Epoch time (integer seconds since
         1970/1/1 00:00:00). Must match time at beginning of fire
         (QU_Simparams.inp and QUIC_fire.inp)
+    wind_times : NonNegativeFloat | list(NonNegativeFloat)
+        Time in seconds since the start of the fire for each wind shift.
+        First time must be zero.
+    wind_speeds : PositiveFloat | list(PositiveFloat)
+        Wind speed or list of wind speeds (m/s)
+    wind_directions : NonNegativeInt < 360 | list(NonNegativeInt < 360)
+        Wind direction or list of directions (degrees). Use 0° for North
     sensor_height : PositiveFloat
         Wind measurement height (m). Default is 6.1m (20ft)
-    wind_speed : PositiveFloat
-        Wind speed (m/s)
-    wind_direction : NonNegativeInt < 360
-        Wind direction (degrees). Use 0° for North
+    x_location : PositiveInt
+        Location of the sensor in the x-direction
+    y_location : PositiveInt
+        Location of the sensor in the y-direction
     """
 
-    name: str = "sensor1"
+    name: str
     _extension: str = ".inp"
     time_now: PositiveInt
+    wind_times: Union[NonNegativeFloat, List[NonNegativeFloat]] = 0
+    wind_speeds: Union[PositiveFloat, List[PositiveFloat]]
+    wind_directions: Union[NonNegativeInt, List[NonNegativeInt]]
     sensor_height: PositiveFloat = 6.1
-    wind_speed: PositiveFloat
-    wind_direction: NonNegativeInt = Field(lt=360)
+    x_location: PositiveInt = 1
+    y_location: PositiveInt = 1
+
+    @field_validator("wind_times", mode="before")
+    @classmethod
+    def validate_wind_times(cls, v: list, values) -> list:
+        if isinstance(v, (float, int)):
+            v = [v]
+        if v[0] != 0:
+            raise ValueError("wind_times: first element of sensor.wind_times must be 0")
+        return v
+
+    @field_validator("wind_speeds", "wind_directions")
+    @classmethod
+    def validate_wind_lists(cls, v, values) -> list:
+        if isinstance(v, (float, int)):
+            v = [v]
+        if v and len(values.data["wind_times"]) != len(v):
+            raise ValueError(
+                f"WindSensor: lists of wind times, speeds, and directions must be the same length.\n"
+            )
+        return v
 
-    @computed_field
     @property
-    def _wind_lines(self) -> str:
+    def _filename(self):
+        return f"{self.name}{self._extension}"
+
+    @property
+    def documentation_dict(self) -> dict:
+        # Return the documentation dictionary
+        with open(DOCS_PATH / f"{self._filename}.json", "r") as f:
+            return json.load(f)
+
+    def _find_last_wind_time(self, target_value):
+        index = bisect.bisect_left(self.wind_times, target_value)
```

Comment:
You're right that this returns the index of the most recent wind time relative to a target value. Since it's only used within another function I didn't think too hard about the naming. What do you think of _find_target_time_index?